rust-runtime/aws-smithy-runtime-api/src/client/http.rs

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! HTTP clients and connectors
+//!
+//! # What is a connector?
+//!
+//! When we talk about connectors, we are referring to the [`HttpConnector`] trait, and implementations of
+//! that trait. This trait simply takes a HTTP request, and returns a future with the response for that
+//! request.
+//!
+//! This is slightly different from what a connector is in other libraries such as
+//! [`hyper`](https://crates.io/crates/hyper). In hyper 0.x, the connector is a
+//! [`tower`](https://crates.io/crates/tower) `Service` that takes a `Uri` and returns
+//! a future with something that implements `AsyncRead + AsyncWrite`.
+//!
+//! The [`HttpConnector`] is designed to be a layer on top of
+//! whole HTTP libraries, such as hyper, which allows Smithy clients to be agnostic to the underlying HTTP
+//! transport layer. This also makes it easy to write tests with a fake HTTP connector, and several
+//! such test connector implementations are available in [`aws-smithy-runtime`](https://crates.io/crates/aws-smithy-runtime)
+//! with the `test-util` feature enabled.
+//!
+//! # Responsibilities of a connector
+//!
+//! A connector primarily makes HTTP requests, but is also the place where connect and read timeouts are
```

Comment:
(and write timeouts, I think)

Reply:
I don't think we have write timeouts.

Reply:
https://github.com/hyperium/hyper/issues/1234
🤔 I'm not sure what the status of 'write timeouts' in `hyper` is.